### ISSUE TITLE
No augmentation on validation split

### ISSUE THREAD

**Issue by Seanspt** [keras-team/keras-preprocessing#218]
When using `flow_from_directory` with subset training and validation,  is it more reasonable to perform ONLY resize and rescale on validation images?

**Comment by rragundez** [comment]
You can set that up with the arguments in `flow_from_directory`.

**Comment by cbernet** [comment]
Hello, I'm sorry to revive this, but I am also hitting this problem. Clearly, I don't want augmentation on the validation dataset because:

* this would underestimate the accuracy.
* I do not want to have artificial fluctuations in the validation accuracy due to random augmentation.

I've studied the code for a while, and I do not see how to disable augmentation in `flow_from_directory` for the validation subset only. 

As far as I can see the augmentation is done [here](https://github.com/keras-team/keras-preprocessing/blob/master/keras_preprocessing/image/iterator.py#L238) and I can't see how this could depend on the subset : `validation` or `training`. 

Could you please tell me exactly how to disable augmentation for the validation subset in `flow_from_directory`? If you confirm that the feature is missing, and if you accept PRs, I'd be happy to contribute. 

Cheers!

**Comment by rragundez** [comment]
Indeed you can create a PR: basically you need something like `... and self.subset == 'training:'` 

https://github.com/keras-team/keras-preprocessing/blob/master/keras_preprocessing/image/iterator.py#L236

Have you looked at other issues that might be related? Perhaps other people have a reason to perform augmentation in the validation set as well.

**Comment by cbernet** [comment]
Thanks for your answer. Yes, something like this would work. It's actually quite simple. 

I'm not sure why anybody would like to perform augmentation on the validation set, but it's indeed possible that people might have good reasons to do so. In fact, what I would like to do is to keep the current behaviour (augmentation on the validation set is ON), and allow users to disable augmentation for this set with an additional  parameter, if you agree. And you can decide to change the default later on if needed. 

I would like to also make a pull request to the documentation to explain this, if that's ok with you. 

If you agree, I'll get started. Thanks again!
